### PR TITLE
1717 add a parameter to spcify amount and limits to with contract call

### PR DESCRIPTION
**Pull request by hui-an-yang**
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: integration-tests/batch-api.spec.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ CONFIGS().forEach(({ lib, rpc, setup, knownBaker, createAddress }) => {
       done()
     })
     it('Verify simple batch transfers with origination', async (done) => {
-      const batch = await Tezos.batch()
+      const batch = await Tezos.contract.batch()
```
Comment (roxaneletourneau) [inline integration-tests/batch-api.spec.ts:15]: We already have a file to test the contract.batch: https://github.com/ecadlabs/taquito/blob/master/integration-tests/contract-batch.spec.ts
This one should not be modified. We will delete it when we remove the deprecated method.

File: packages/taquito/src/batch/rpc-batch-provider.ts [hunk 2]
```diff
@@ -94,14 +95,16 @@ export class OperationBatch extends OperationEmitter {
 
   /**
    *
-   * @description Add a transaction operation to the batch
+   * @description Add a coantract call to the batch
```
Comment (roxaneletourneau) [inline packages/taquito/src/batch/rpc-batch-provider.ts:98]: typo in contract

**Review by roxaneletourneau — CHANGES_REQUESTED** [review]
Looks good :) some minor comments

File: integration-tests/batch-api.spec.ts (hunk 1 above)
Comment (hui-an-yang) [reply integration-tests/batch-api.spec.ts]: Gotcha, it's been rolled back

File: packages/taquito/src/batch/rpc-batch-provider.ts (hunk 2 above)
Comment (hui-an-yang) [reply packages/taquito/src/batch/rpc-batch-provider.ts]: Thanks, pushed new commit to address this

**Review by roxaneletourneau — APPROVED** [review]

**Pull request merged**